core/src/types/operator/registry.rs

```diff
@@ -0,0 +1,226 @@
+use std::cell::LazyCell;
+use std::collections::HashMap;
+
+use crate::services::*;
+use crate::*;
+
+// TODO: thread local or use LazyLock instead? this way the access is lock-free
+// TODO: should we expose the `GLOBAL_OPERATOR_REGISTRY` as public API at `crate::types::operator::GLOBAL_OPERATOR_REGISTRY`?
+thread_local! {
```

Comment:
What is the preferred way of having a global static variable such as this?

I prefer to have it `thread_local` so there is not need for a `LazyLock`, we can use `LazyCell` instead (`LazyCell` is lock-free but `LazyLock` is not, it synchronizes access through threads)